src/processor/operator/persistent/rel_batch_insert.cpp

```diff
@@ -225,5 +233,11 @@ void RelBatchInsert::finalizeInternal(ExecutionContext* context) {
     partitionerSharedState->partitioningBuffers[relInfo->partitioningIdx].reset();
 }
 
+void RelBatchInsert::updateProgress(ExecutionContext* context) {
+    double progress =
```

Comment:
Though not likely to happen, but I think you should guard against the cast when `progressSharedState->partitionsTotal` is `0`.